src/main/haskell/language-kore/src/Data/Kore/Unparser/Unparse.hs

```diff
@@ -108,18 +111,36 @@ instance Unparse (Variable level) where
     unparse var =
         unparse (variableName var) >> write ":" >> unparse (variableSort var)
 
-instance Unparse UnifiedSortVariable where
-    unparse (ObjectSortVariable sv) = unparse sv
-    unparse (MetaSortVariable sv)   = unparse sv
+instance
+    ( Unparse (sort Object)
+    , Unparse (sort Meta)
+    ) => Unparse (Unified sort)
+  where
+    unparse (UnifiedObject s) = unparse s
+    unparse (UnifiedMeta s)   = unparse s
 
-instance Unparse (UnifiedVariable Variable) where
-    unparse (ObjectVariable sv) = unparse sv
-    unparse (MetaVariable sv)   = unparse sv
+instance
```

Comment:
I somehow feel that one should be able to descend directly from the top of {PatternObjectMeta, UnifiedPattern, Fix, Pattern level} to the bottom, without going through all the intermediate levels.